msf/msf.go

```diff
@@ -0,0 +1,260 @@
+					return xerrors.Errorf("error in save: %w", err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// WalkDirTree :
+func WalkDirTree(root string) ([]models.Module, error) {
+	modules := []models.Module{}
+
+	err := filepath.Walk(root,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return xerrors.Errorf("file walk error: %w", err)
+			}
+			if info.IsDir() {
+				return nil
+			}
+
+			// Exclude the sample code
+			sample, err := filepath.Match("example*", info.Name())
+			if err != nil {
+				return xerrors.Errorf("error in file check: %w", err)
+			}
+			if sample {
+				return nil
+			}
+
+			f, err := ioutil.ReadFile(path)
+			if err != nil {
+				return xerrors.Errorf("error in file open: %w", err)
+			}
+
+			module, err := Parse(f, path)
+			if err != nil {
+				return xerrors.Errorf("error in parse: %w", err)
+			}
+			modules = append(modules, *module)
+
+			return nil
+		})
+	if err != nil {
+		return nil, xerrors.Errorf("error in walk: %w", err)
+	}
+
+	return modules, nil
+}
+
+// Parse : Extracts information from update_info of module as a regular expression.
+func Parse(file []byte, path string) (*models.Module, error) {
+	// Title
+	var title string
+	titleMatches := titleRegex.FindAllSubmatch(file, -1)
+	// Substitute the first of the matched elements into the title
+	for _, m := range titleMatches {
+		if 1 < len(m) {
+			title = fmt.Sprintf("%s", m[1])
+			break
+		}
+	}
+
+	// Description
+	var decs string
+	decsRegxps := []*regexp.Regexp{
+		summaryRegexp1,
+		summaryRegexp2,
+		summaryRegexp3,
+	}
+	// Substitute the first of the matched elements into the decs
+	for _, re := range decsRegxps {
+		decsMatches := re.FindAllSubmatch(file, -1)
+		for _, m := range decsMatches {
+			if 1 < len(m) {
+				decs = formatDescription(m)
+				if decs != "" {
+					break
+				}
+			}
+		}
+		if decs != "" {
+			break
+		}
+	}
+
+	// Cve id
+	var cveIDs []string
+	cveMatches := cveIDRegexp.FindAllSubmatch(file, -1)
+	for _, m := range cveMatches {
+		if 2 < len(m) {
+			cveID := fmt.Sprintf("CVE-%s-%s", m[1], m[2])
+			cveIDs = append(cveIDs, cveID)
+		}
+	}
+
+	// Exploitdb unique id
+	var edbIDs []string
+	edbMatches := edbIDRegexp.FindAllSubmatch(file, -1)
+	for _, m := range edbMatches {
+		if 1 < len(m) {
+			edbID := fmt.Sprintf("EDB-%s", m[1])
+			edbIDs = append(edbIDs, edbID)
+		}
+	}
+
+	// Referenses
+	var links []string
+	urlRegxps := []*regexp.Regexp{
+		cweRegexp,
+		bidRegexp,
+		zdiRegexp,
+		msbRegexp,
+		wpvdbRegexp,
+		uscertRegexp,
+		packetRegexp,
+		refRegexp,
+	}
+	for _, re := range urlRegxps {
+		urlMatches := re.FindAllSubmatch(file, -1)
+		for _, m := range urlMatches {
+			url := formatReferences(m)
+			links = append(links, url)
+		}
+	}
+
+	return &models.Module{
+		Name:        filepath.Base(path),
+		Title:       title,
+		Description: decs,
+		CveIDs:      cveIDs,
+		EdbIDs:      edbIDs,
+		References:  links,
+	}, nil
+}
+
+func formatDescription(match [][]byte) string {
```

Comment:
Please add a test case for this function.

Reply:
This function is included in the Parse test.   
Should I set up a separate test case to check it?